### ISSUE TITLE
Bug: Adding multiple rules to `eslint-disable-next-line` along with `noInlineConfig: true` somehow doesn't report any errors

### ISSUE THREAD

**Issue by ankitprahladsoni** [eslint/eslint#20344]
### Environment

Node version: 22.14.0
npm version: 11.6.2
Local ESLint version: 9.30
Operating System: OSX

### What parser are you using?

Default (Espree)

### What did you do?

`eslint-config.js`
```js
import js from "@eslint/js";
import globals from "globals";
import reactHooks from "eslint-plugin-react-hooks";
import reactRefresh from "eslint-plugin-react-refresh";
import tseslint from "typescript-eslint";
import { defineConfig, globalIgnores } from "eslint/config";

export default defineConfig([
  globalIgnores(["dist"]),
  {
    files: ["**/*.{ts,tsx}"],
    extends: [
      js.configs.recommended,
      tseslint.configs.recommended,
      reactHooks.configs.flat.recommended,
      reactRefresh.configs.vite,
    ],
    languageOptions: {
      ecmaVersion: 2020,
      globals: globals.browser,
    },
    linterOptions: {
      noInlineConfig: true,
    },
  },
]);

```

`App.tsx`
```tsx
import { useEffect, useState } from "react";
import "./App.css";

function App() {
  const [count, setCount] = useState(0);

  useEffect(() => {
    // eslint-disable-next-line react-hooks/exhaustive-deps, react-hooks/set-state-in-effect
    setCount(count + 1);
  }, []);

  return <div>Test</div>;
}

export default App;

```

### What did you expect to happen?

ESLint should have thrown at least one error, since `noInlineConfig` is set to `true`. We can get it if we remove the inline config manually
`8:5 error    Error: Calling setState synchronously within an effect can trigger cascading renders`

### What actually happened?

Running `npm run eslint` results in 2 warnings
```
   8:5  warning  '// eslint-disable-next-line react-hooks/exhaustive-deps, react-hooks/set-state-in-effect' has no effect because you have 'noInlineConfig' setting in your config
  10:6  warning  React Hook useEffect has a missing dependency: 'count'. Either include it or remove the dependency array. You can also do a functional update 'setCount(c => ...)' if you only need 'count' in the 'setCount' call  react-hooks/exhaustive-deps
```

### Link to Minimal Reproducible Example

https://github.com/ankitprahladsoni/eslint-noInlineConfig-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Thanks for the issue and the repro. It took me a while to figure out what was causing this behavior that looks like a bug in ESLint but is actually caused by `eslint-plugin-react-hooks` searching for eslint directives in the source code and altering behavior of rules if certain directives are found.

I'm not sure where exactly this code is in the plugin source code, but I can see it in `node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js` starting from line 53021:

```js
function findProgramSuppressions(programComments, ruleNames, flowSuppressions) {
    const suppressionRanges = [];
    let disableComment = null;
    let enableComment = null;
    let source = null;
    let disableNextLinePattern = null;
    let disablePattern = null;
    let enablePattern = null;
    if (ruleNames.length !== 0) {
        const rulePattern = `(${ruleNames.join('|')})`;
        disableNextLinePattern = new RegExp(`eslint-disable-next-line ${rulePattern}`);
        disablePattern = new RegExp(`eslint-disable ${rulePattern}`);
        enablePattern = new RegExp(`eslint-enable ${rulePattern}`);
```

When I modify `disableNextLinePattern` to a pattern that doesn't match disable directives, lint error `Error: Calling setState synchronously within an effect can trigger cascading renders` appears as expected. So the unexpected behavior isn't caused by ESLint  but by `eslint-plugin-react-hooks`. You can open an issue in the plugin repo (https://github.com/facebook/react).
